File: DI/DI-Dagger-Java/src/main/java/DI/named/MyAppComponent.java
package DI.named;

import dagger.Component;
import model.named.MyApp;

import javax.inject.Singleton;

/**
 * Siempre puedes llamar a esto como más te guste, como quieras...
 * para luego recuperarlo como.
 */
@Singleton
@Component(modules = {
        AppServiceModule.class,
        AppCollaboratorModule.class
})
public interface MyAppComponent {
    MyApp build(); // También puedes llamarlo como quieras

    // O simplemente si no hacemos la inyección en el constructor
    // Esto es interesante en sitios donde no hay cosntructores
    // Como Android y sus actividades
    // void inject(MyApp Class);
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/DI/named/MyAppComponent.kt
package DI.named

import dagger.Component
import models.named.MyApp
import javax.inject.Singleton

@Singleton
@Component(
    modules =
    [
        AppServiceModule::class,
        AppCollaboratorModule::class
    ]
)
interface MyAppComponent {
    fun build(): MyApp // También puedes llamarlo como quieras
    // O simplemente si no hacemos la inyección en el constructor
    // Esto es interesante en sitios donde no hay cosntructores
    // Como Android y sus actividades
    // void inject(MyApp Class);
}
